viewing_party/party.py

```diff
@@ -1,23 +1,201 @@
+        
+    for movie in watched:
+        if genre_dict[movie["genre"]] >= 1:
+            genre_dict[movie["genre"]] += 1
+
+
+    # max(d, key = d.get)
+    most_watched = max(genre_dict, key = genre_dict.get)
+    
+    return most_watched
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+
+# Consider the movies that the user has watched, 
+# and consider the movies that their friends have watched. 
+# Determine which movies the user has watched, 
+# but none of their friends have watched.
+# Return a list of dictionaries, that represents a list of movies
+
+    user_watched = user_data["watched"]
+    friends = user_data["friends"]
+
+    user_unique_movies = []
+
+    friends_movie_watched = []
+
+    # Big O notation of O(n^2), try to refactor
+    for friend in friends:
+        for i in range(len(friend["watched"])):
+            friends_movie_watched.append(friend["watched"][i]) 
+
+    for movie in user_watched:
+        if movie not in friends_movie_watched:
+            if movie not in user_unique_movies:
+                user_unique_movies.append(movie)
+
+    return user_unique_movies
+
+def get_friends_unique_watched(user_data):
+    
+# Consider the movies that the user has watched, 
+# and consider the movies that their friends have watched. 
+# Determine which movies at least one of the user's friends have watched, 
+# but the user has not watched.
+# Return a list of dictionaries, that represents a list of movies
+    
+    user_watched = user_data["watched"]
+    friends = user_data["friends"]
+
+    friends_unique_movies = []
+
+    friends_movie_watched = []
+
+    # Big O notation of O(n^2), try to refactor
+    for friend in friends:
+        for i in range(len(friend["watched"])):
+            friends_movie_watched.append(friend["watched"][i]) 
+
+    for movie in friends_movie_watched:
+        if movie not in user_watched:
+            if movie not in friends_unique_movies:
+                friends_unique_movies.append(movie)
+
+    return friends_unique_movies
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+
+# Determine a list of recommended movies. 
+# A movie should be added to this list if and only if:
+# - The user has not watched it
+# - At least one of the user's friends has watched
+# - The `"host"` of the movie is a service that is in 
+# - the user's `"subscriptions"`
+# Return the list of recommended movies
+
+    recommendations = []
+    
+    for movie in get_friends_unique_watched(user_data):
+        if movie["host"] in user_data["subscriptions"]:
+            recommendations.append(movie)
+
+    return recommendations
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------

```

Comment:
Your wave 5 was also concise and nicely done! Couldn't have done it better myself!